lib/axiom/fs/nested_stdio.js

```diff
@@ -45,29 +45,31 @@ export default NestedStdio;
 NestedStdio.prototype = Object.create(Stdio.prototype);
 
 /**
- * End the stream forwarders so that the .
+ * Calls `end` on all stream forwarders.
  *
  * @return {void}
  */
 NestedStdio.prototype.end = function() {
-  this.stdin.end();
-  this.signal.end();
-  this.ttyin.end();
-  this.stdout.end();
-  this.stderr.end();
-  this.ttyout.end();
+  for(var key in this) {
+    var item = this[key];
+    if ((item instanceof ReadableStreamForwarder) ||
+        (item instanceof WritableStreamForwarder)) {
+      item.end();
+    }
+  }
 };
 
 /**
- * Detach event handlers of the stream forwarders from their source streams.
+ * Calls `close` on all stream forwarders.
  *
  * @return {void}
  */
 NestedStdio.prototype.close = function(error) {
-  this.stdin.close(error);
-  this.signal.close(error);
-  this.ttyin.close(error);
-  this.stdout.close(error);
-  this.stderr.close(error);
-  this.ttyout.close(error);
+  for(var key in this) {
```

Comment:
This pattern is a little too shotgun, I think.  Maybe

    ['signal', ..., 'ttyout'].forEach(function(key) {
     ...
    });

would feel less dangerous.